### PR TITLE
tests for api/osd/configure/ and ceph_installer.process.make_ansible_command

### PR DESCRIPTION
**Pull request by andrewschoen**

### CODE REVIEW

File: ceph_installer/tests/controllers/test_osd.py [hunk 1]
```diff
@@ -3,18 +3,10 @@
...
-        data = dict()
-        result = session.app.post_json("/api/osd/configure/", params=data,
-                                       expect_errors=True)
-        assert result.status_int == 400
+    def setup(self):
+        self.configure_data = self.get_configure_data()
 
-    def test_configure_success(self, session, monkeypatch):
-        monkeypatch.setattr(osd.call_ansible, 'apply_async', lambda args, kwargs: None)
+    def get_configure_data(self):
```
Comment (alfredodeza) [inline ceph_installer/tests/controllers/test_osd.py]: why is this not part of setup() anymore? why make it a method?

File: ceph_installer/tests/test_process.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+from ceph_installer import process
+
+
+class TestMakeAnsibleCommand(object):
+
+    def test_with_tags(self, monkeypatch):
+        monkeypatch.setattr(process, 'which', lambda x: "/bin/ansible")
```
Comment (alfredodeza) [inline ceph_installer/tests/test_process.py]: yesssss monkeypatch

File: ceph_installer/tests/controllers/test_osd.py (hunk 1 above)
Comment (andrewschoen) [inline ceph_installer/tests/controllers/test_osd.py]: good call; fixed.

File: ceph_installer/tests/controllers/test_osd.py [hunk 3]
```diff
@@ -26,8 +15,77 @@ def test_configure_success(self, session, monkeypatch):
...
+        def check(args, kwargs):
+            extra_vars = kwargs["extra_vars"]
+            assert "redhat_storage" not in extra_vars
+            assert "ceph_stable_rh_storage" in extra_vars
+
+        data = self.configure_data.copy()
+        data["redhat_storage"] = True
+        monkeypatch.setattr(osd.call_ansible, 'apply_async', check)
+        result = session.app.post_json("/api/osd/configure/", params=data,
+                                       expect_errors=True)
```
Comment (alfredodeza) [inline ceph_installer/tests/controllers/test_osd.py]: why are you expecting errors if the assert is for a 200?

File: ceph_installer/tests/controllers/test_osd.py [hunk 4]
```diff
@@ -26,8 +15,77 @@ def test_configure_success(self, session, monkeypatch):
...
+        assert result.status_int == 200
+
+    def test_configure_playbook(self, session, monkeypatch):
+        def check(args, kwargs):
+            assert "osd-configure.yml" in kwargs["playbook"]
+
+        monkeypatch.setattr(osd.call_ansible, 'apply_async', check)
+        result = session.app.post_json("/api/osd/configure/", params=self.configure_data,
+                                       expect_errors=True)
```
Comment (alfredodeza) [inline ceph_installer/tests/controllers/test_osd.py]: same expectation. Seems to me you don't want this

**Comment by andrewschoen** [comment]
test this please

File: ceph_installer/tests/controllers/test_osd.py (hunk 3 above)
Comment (andrewschoen) [inline ceph_installer/tests/controllers/test_osd.py]: I should not be expecting errors, I've got this fixed now.

**Pull request merged**